tests/Schema/Keywords/PatternTest.php

```diff
@@ -10,16 +10,33 @@
 
 final class PatternTest extends SchemaValidatorTest
 {
-    public function testItValidatesPatternGreen() : void
+    /**
+     * @return array<array<string, string>>
+     */
+    public function validDataProvider() : array
+    {
+        return [
+            ['^[a|b]+$', 'abba'],
+            ['foo', 'foo'], // Tests adding anchors
+            ['foof', 'foof'], // Tests adding anchors when first and last character is same
```

Comment:
not really a problem, but IMO having array keys instead of (or alongside with) comments is a bit better since then phpunit gives you this key as a failed test reference instead of numeric order of dataset, and it helps a lot.

Also I think comments are now a bit overcomplicated, since we are not testing whether we should add anchors or not as before, but just test various edge-cases